Code/Framework/AzCore/AzCore/Math/Uuid.inl

```diff
@@ -0,0 +1,487 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#include <AzCore/Math/Sha1.h>
+
+#include <AzCore/Casting/numeric_cast.h>
+#include <AzCore/std/containers/array.h>
+#include <AzCore/std/limits.h>
+#include <AzCore/std/ranges/ranges_algorithm.h>
+#include <AzCore/std/ranges/ranges_to.h>
+#include <AzCore/std/string/fixed_string.h>
+
+namespace AZ::UuidInternal
+{
+    constexpr AZStd::byte GetValue(char c)
+    {
+        constexpr AZStd::string_view UuidChars = "0123456789ABCDEFabcdef";
+        // CharToHexDigit is 1 more character than UuidChars array
+        constexpr auto CharToHexDigit = AZStd::to_array<AZStd::byte>({
+            AZStd::byte(0),AZStd::byte(1), AZStd::byte(2), AZStd::byte(3),
+            AZStd::byte(4), AZStd::byte(5), AZStd::byte(6), AZStd::byte(7),
+            AZStd::byte(8), AZStd::byte(9),
+            AZStd::byte(10), AZStd::byte(11), AZStd::byte(12), AZStd::byte(13), AZStd::byte(14), AZStd::byte(15),
+            AZStd::byte(10), AZStd::byte(11), AZStd::byte(12), AZStd::byte(13), AZStd::byte(14), AZStd::byte(15),
+            AZStd::numeric_limits<AZStd::byte>::max() });
+        const char* hexDigit = AZStd::ranges::find(UuidChars, c);
+        return CharToHexDigit[hexDigit - UuidChars.data()];
+    }
+}
+
+namespace AZ
+{
+    constexpr Uuid::Uuid(AZStd::string_view uuidString)
+    {
+        operator=(CreateString(uuidString));
+    }
+    constexpr Uuid::Uuid(const char* string, size_t stringLength)
+    {
+        operator=(CreateString(string, stringLength));
+    }
+
+    constexpr Uuid Uuid::CreateNull()
+    {
+        return Uuid{};
+    }
+
+    constexpr Uuid Uuid::CreateString(const char* string, size_t stringLength)
+    {
+        return CreateString({ string, stringLength });
+    }
+
+    constexpr Uuid Uuid::CreateString(AZStd::string_view uuidString)
+    {
+        return CreateStringSkipWarnings(uuidString, false);
+    }
+
+    constexpr Uuid Uuid::CreateStringSkipWarnings(AZStd::string_view uuidString, [[maybe_unused]] bool skipWarnings)
+    {
+        if (uuidString.empty())
+        {
+            return {};
+        }
+
+        if (uuidString.size() < 32 || uuidString.size() > 38)
+        {
+            AZStd::string_view errorString = !uuidString.empty() ? uuidString : AZStd::string_view("null");
+            AZ_Warning("Math", skipWarnings, "Invalid UUID format %.*s (must be) {xxxxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx} (or without dashes and braces)",
```

Comment:
This Warning format string is repeated 2 more times below, maybe define it to a variable.

Reply:
Will do.
Hopefully this doesn't trigger any warnings about format strings not being string literal.
Well we will find out.